---
title: "MIDY_adipose_data_with_two_missing_values_filtered"
author: "Erik Ländström"
date: "14 December 2018"
output: 
  html_document:
    keep_md: TRUE
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries

```{r library, message = FALSE, warning = FALSE}
library(tidyverse)
library(DEP)
library(conflicted)
library(biomaRt)
library(naniar)
library(knitr)
library(readxl)
library(gridExtra)
library(visdat)
library(qvalue)
library(modelr)
```


# 2. Preprocessing

## 2.1 Read rawdata from Maxquant

```{r, }
make_tibble_for_DEP <- function(tb = "ProteinGroups.txt") {
  data_raw <- read_tsv(tb)
  lfq_col_names <- str_extract(colnames(data_raw), "LFQ.+")
  lfq_col_names <- lfq_col_names[which(!is.na(lfq_col_names))]
  
  data_raw <<- data_raw %>%
  dplyr::select(`Protein IDs`, `Majority protein IDs`,
                `Fasta headers`, Peptides, 
                `Razor + unique peptides`, `Unique peptides`, 
                lfq_col_names,
                `Only identified by site`, `Reverse`,
                `Potential contaminant`)
}
```

```{r message = FALSE}
make_tibble_for_DEP("proteinGroups.txt")
```

## 2.2 Remove contaminant and reverse hits

Calculate number of contaminants and reverse hits.

```{r}
data_raw %>%
  summarise(n_reverse = length(Reverse) - sum(is.na(Reverse)),
            n_contaminant = length(Reverse) - sum(is.na(`Potential contaminant`)))
```

```{r}
remove_contaminants_and_reverse_hits <- function(tb) {
  data_raw <<- tb %>%
    dplyr::filter(is.na(`Potential contaminant`)) %>%
    dplyr::filter(is.na(Reverse))
}
```

```{r}
remove_contaminants_and_reverse_hits(data_raw)
dim(data_raw)
```

## 2.3 Add gene names from BiomaRt

```{r scripts}
# Remove isoform from refseq identifier
data_raw <- data_raw %>%
  mutate(Protein_ID = str_extract(`Majority protein IDs`, "NP_[0-9]+|XP_[0-9]+"))
```

Extract gene names from biomart.

```{r}
convert_NCBIIDs_to_gene_names <- function(tb) {
  
  # Load sus scrofa dataset from biomaRt and change column name to Protein_ID
  ensembl <- useMart("ensembl", dataset = "sscrofa_gene_ensembl")  
  
  # Get gene names for peptide IDs
  NP <- as_tibble(getBM(attributes = c("refseq_peptide", "external_gene_name"),
              filters = "refseq_peptide",
              values = tb,
              mart = ensembl
              ))
  
  NP <- NP %>%
    rename(Protein_ID = refseq_peptide)
  
  # Load sus scrofa dataset from biomaRt and change column name to Protein_ID
  XP <- as.tibble(getBM(attributes = c("refseq_peptide_predicted", "external_gene_name"),
              filters = "refseq_peptide_predicted",
              values = tb,
              mart = ensembl
              ))
  
  XP <- XP %>%
    rename(Protein_ID = refseq_peptide_predicted)
  
  # Use full_join to join the two tables
  gene_names <- full_join(NP, XP, by = "Protein_ID")
  
  # Tidy data, gene names in one column and remove NAs
  gene_names <- gene_names %>%
    gather(external_gene_name.x, external_gene_name.y, key = "Source", value = "Gene_name") %>%
    dplyr::select(Protein_ID, Gene_name) %>%
    dplyr::filter(!is.na(Gene_name)) 
  
  # If there are any "" artifacts from the biomaRt search, replace them with NA
  gene_names[gene_names == ""] <- NA
  
  # Save output in a tibble
  gene_names <<- gene_names
}

convert_NCBIIDs_to_gene_names(data_raw$Protein_ID)
```

Join gene names with raw data by using protein IDs as identifiers.

```{r join_genenames}
data <- full_join(data_raw, gene_names, by = "Protein_ID")
```

Check if there are any duplicated gene names and identifiers.

```{r unique}
# gene names
data$Gene_name %>% duplicated() %>% any()

# identifiers
data$Protein_ID %>% duplicated() %>% any()
```

Duplicated gene names were found in the data.

```{r unique2}
# Number of NA gene names
data %>%
  summarise(sum(is.na(Gene_name)))

# Count number of instances a gene name appears
data %>%
  group_by(Gene_name) %>%
  summarise(n = n()) %>%
  dplyr::filter(n > 1) %>%
  arrange()

# Make unique "row names", either gene name or identifier
data_unique <- make_unique(data, "Gene_name", "Protein_ID")

data_unique %>%
  dplyr::select(name, ID) %>% 
  duplicated() %>%
  any()
```

## 2.4 Generate a SummarizedExperiment

```{r SExperiment}
# Grep for lfq columns
lfq_columns <- grep("LFQ.", colnames(data_unique))

# Read experimental design
experimental_design <- read_xlsx("MIDY_adipose_experimental_design.xlsx")

# Generate summarized experiment
data_se <- make_se(data_unique, lfq_columns, experimental_design)

data_se
```

# 3. Analysis

## 3.1 Missing value filtering

```{r frequency}
plot_frequency(data_se)
```


```{r}
# Filter for proteins present in all samples of at least one condition
data_2_valid_values <- filter_missval(data_se, thr = 3)
```

```{r impute}
# Imputation
imputed_data <- impute(data_2_valid_values, fun = "man", shift = 1.8, scale = 0.3)
```

```{r}
# Save imputed data in a tibble
adipose_imputed <- get_df_long(imputed_data)

# Tidy up the condition column
adipose_imputed <- adipose_imputed %>%
  separate(condition, into = c("tissue", "genotype"), sep = "_") %>%
  rename(LFQ = intensity)
```





# 2-way ANOVA

Perform a 2-way ANOVA on the imputed dataset.

```{r echo = FALSE}
multiple_2way_anova_in_tidyverse_with_tukeyHSD <- function(tb, cases, observations,
                                                           independent_var1,
                                                           independent_var2) {
  
  # Libraries
  library(tidyverse)
  library(broom)
  
  # Quoting
  cases <- enquo(cases)
  
  lm_formula <- as.formula(paste(observations, "~", independent_var1, "*", independent_var2))
  
  interaction_effect <- paste(independent_var1, ":", independent_var2, sep = "")
  
  # 2-way ANOVA
  anova <- tb %>%
    group_by(!!cases) %>%
    nest() %>%
    mutate(anova = map(data, ~ aov(lm_formula, data = .x)),
           tukey = map(data, ~ TukeyHSD(aov(lm_formula, data = .x),
                                        which = interaction_effect)),
           tidy_anova = map(anova, tidy),
           tidy_tukey = map(tukey, tidy))
  
  return(anova)
}
```

```{r anova}
adipose_anova <- multiple_2way_anova_in_tidyverse_with_tukeyHSD(adipose_imputed, name, "LFQ", "genotype", "tissue")
```

Unnest anova list-column.

```{r echo = FALSE}
make_significant_ANOVA_kable <- function(tb) {
    
  kable(tb %>%
          dplyr::filter(term != "Residuals") %>%
          group_by(term) %>%
          summarise(n = sum(p.value < 0.05)),
        caption = "Number of significant differences per factor.", format = "pandoc",
        align = "c")
}

plot_pvalue_histogram_ANOVA <- function(tb, col_name, limit_max) {
  
  # Quote
  col_name <- enquo(col_name)
  
  # pvalue plot
  tb %>% 
    ggplot(aes(x = !!col_name)) +
    geom_histogram(breaks = seq(0, 1, 0.05), fill = "black", color = "white") +
    xlab("p-value") +
    ylab("Number of proteins") +
    facet_wrap(~ term) +
    scale_y_continuous(limits = c(0, limit_max), expand = c(0, 0)) + 
    labs(
      title = "p-value histograms for each ANOVA factor"
    ) +
    theme(panel.background = element_blank(),
          axis.line = element_line(color = "black"))
}
```

```{r anova_results}
# Extract p-values and more
anova_results <- adipose_anova %>%
  unnest(tidy_anova) %>% 
  dplyr::filter(term != "Residuals")

anova_results

# Count number of significant p-values
make_significant_ANOVA_kable(anova_results)

# Plot pvalue histograms for each factor
plot_pvalue_histogram_ANOVA(anova_results, p.value, 750)
```

```{r}
# Function
fdr_correction_with_qvalue <- function(tb, p_value_col) {
  
  # Quote
  p_value_col <- enquo(p_value_col)
  
  # fdr correction
  qvalues <- qvalue(as_vector(tb %>%
                                ungroup() %>% # To make sure to only get one column
                                dplyr::select(!!p_value_col)))
  
  return(qvalues)
}

# fdr correction with the qvalue package
qvalues <- fdr_correction_with_qvalue(anova_results, p.value)

# Plot fdr statistics
plot(qvalues)
hist(qvalues)

# Function
join_qvalues_with_ANOVA_results <- function(tb) {
  qvalues <- tibble(p.value = qvalues[["pvalues"]], qvalue = qvalues[["qvalues"]])
  
  anova_results <- full_join(anova_results, qvalues, by = "p.value")
  return(anova_results)
}

# Join qvalues with anova results
anova_results <- join_qvalues_with_ANOVA_results(qvalues)

kable(anova_results %>%
        dplyr::filter(term != "Residuals") %>%
        group_by(term) %>%
        summarise(n = sum(qvalue < 0.05)),
      caption = "Number of significant qvalues.", format = "pandoc")

kable(anova_results %>%
        dplyr::filter(term != "Residuals") %>%
        group_by(term) %>%
        summarise(n = sum(qvalue < 0.10)),
      caption = "Number of significant q-values.", format = "pandoc")

```



The p-value histograms doesn't look much different, but with fewer p-values to
correct maybe more q-values will be significant.

```{r no_interaction2}
# fdr correction with the qvalue package
qvalues <- fdr_correction_with_qvalue(anova_results_without_interaction, p.value)

# Plot fdr statistics
plot(qvalues)
hist(qvalues)

# Join qvalues with anova results
anova_results_without_interaction <- join_qvalues_with_ANOVA_results(anova_results_without_interaction)

kable(anova_results_without_interaction %>%
        dplyr::filter(term != "Residuals") %>%
        group_by(term) %>%
        summarise(n = sum(qvalue < 0.05)),
      caption = "Number of significant qvalues.", format = "pandoc")

kable(anova_results_without_interaction %>%
        dplyr::filter(term != "Residuals") %>%
        group_by(term) %>%
        summarise(n = sum(qvalue < 0.10)),
      caption = "Number of significant q-values.", format = "pandoc")
```

When using no interaction more q-values are deemed significant than when an
interaction factor is used.

## FDR correction using `fdrtool`

I want to see what the difference will be when using the `fdrtool` package 
instead of the `qvalue` package for generating FDR.

```{r}
library(fdrtool)

# Calculate qvalues uisng the fdrtool package
fdr_qvalues <- fdrtool(as_vector(anova_results %>%
                    dplyr::select(p.value)),
        statistic = "pvalue")

# Create a tibble with pvalues and qvalues
fdr_qvalues <- tibble(
  p.value = fdr_qvalues[["pval"]],
  fdrtool_qvalue = fdr_qvalues[["qval"]],
  local_FDR = fdr_qvalues[["lfdr"]]
)

# Join fdrtool results with previous anova results (qvalues generated with qvalue)
anova_results <- full_join(anova_results, fdr_qvalues, by = "p.value")

# Print table with significant differences
kable(anova_results %>%
        dplyr::filter(term != "Residuals") %>%
        group_by(term) %>%
        summarise(n_fdrtool = sum(fdrtool_qvalue < 0.05),
                  n_qvalue = sum(qvalue < 0.05)),
      caption = "Number of significant qvalues at < 0.05.", format = "pandoc",
      align = "c")

# Plot qvalues from the two fdr packages to see correlation
anova_results %>% 
  ggplot(aes(qvalue, fdrtool_qvalue)) +
  geom_point() +
  geom_smooth(method = "lm") +
  facet_wrap(~ term)
```

There doesn't seem to be much difference between the results between the two
packages, I will stick to using the `qvalue` package right now.

# ANOVA, further results

I want to explore all the data that the ANOVA outputs.

```{r}
adipose_anova_without_interaction[1, ] %>% 
  dplyr::select(tidy_anova) %>%
  unnest()

adipose_anova_without_interaction[1, ] %>% 
  dplyr::select(augment_anova) %>%
  unnest()

adipose_anova_without_interaction[1, ] %>% 
  dplyr::select(glance_anova) %>%
  unnest()
```

## Plotting example with RDH16
```{r}
rdh16_data <- adipose_imputed %>%
  dplyr::filter(name == "RDH16")

rdh16_data %>% 
  ggplot(aes(genotype, LFQ, color = tissue)) +
  geom_boxplot() 

rdh16_data %>% 
  ggplot(aes(genotype, LFQ)) +
  geom_boxplot()

rdh16_genotype_mean <- rdh16_data %>%
  group_by(genotype) %>% 
  summarise(mean = mean(LFQ))

rdh16_data %>% 
  ggplot() +
  geom_point(aes(genotype, LFQ, color = tissue))+
  geom_point(data = rdh16_genotype_mean, aes(genotype, mean))
```

### Linear model with RDH16 data

```{r}
rdh16_data <- rdh16_data %>% 
  mutate(tissue = as_factor(tissue),
         genotype = as_factor(genotype))

rdh16_model <- lm(LFQ ~ genotype * tissue - 1,data = rdh16_data)
rdh16_model <- lm(LFQ ~ genotype * tissue,data = rdh16_data, subset = genotype == "wt")

rdh16_model <- lm(LFQ ~ genotype - 1, data = rdh16_data)
rdh16_model
rdh16_model$residuals
summary(rdh16_model)
tidy(rdh16_model)
augment(rdh16_model)
glance(rdh16_model)

rdh16_data %>% 
  gather_residuals(rdh16_model) %>% 
  ggplot(aes(tissue, resid)) +
  geom_boxplot()

rdh16_model <- lm(LFQ ~ tissue - 1, data = rdh16_data)
rdh16_model
rdh16_model$residuals
summary(rdh16_model)
tidy(rdh16_model)
augment(rdh16_model)
glance(rdh16_model)

rdh16_data %>% 
  gather_residuals(rdh16_model) %>% 
  ggplot(aes(genotype, resid)) +
  geom_boxplot()
```

## Calculate mean

```{r}
adipose_means <- adipose_imputed %>% 
  group_by(name, tissue, genotype) %>% 
  summarise(
    mean = mean(LFQ)
  ) %>% 
  unite(group, tissue, genotype, sep = "_") %>%
  spread(key = "group", value = "mean") %>% 
  mutate(sc_diff = sc_midy - sc_wt,
         m_diff  = m_midy  - m_wt)

adipose_means %>% 
  ggplot(aes(m_wt, m_midy, color = sc_diff)) +
  geom_point() +
  scale_color_viridis_c()

adipose_means %>% 
  ggplot(aes(sc_diff, m_diff)) +
  geom_point()

tissue_means <- adipose_imputed %>% 
  group_by(name, tissue) %>% 
  summarise(
    mean = mean(LFQ)
  ) %>% 
  spread(key = "tissue", value = "mean")

genotype_means <- adipose_imputed %>% 
  group_by(name, genotype) %>% 
  summarise(
    mean = mean(LFQ)
  ) %>% 
  spread(key = "genotype", value = "mean")

# join mean tables
adipose_means <- full_join(adipose_means[, 1:5], tissue_means, by = "name")
adipose_means <- full_join(adipose_means, genotype_means, by = "name")
```

## Volcano plots

```{r}
# Calculate lfc2 for each group
adipose_lfc2 <- adipose_means %>% 
  group_by(name) %>% 
  mutate(tissue = m - sc,
         genotype = midy - wt) %>%
  dplyr::select(name, tissue, genotype) %>% 
  gather(tissue, genotype, key = "term", value = "lfc2") # term is used for easy joining with anova results

# Join with anova results
anova_results_without_interaction <- full_join(anova_results_without_interaction, adipose_lfc2,
                                               by = c("name", "term")) %>% 
  rename(group = term)

anova_results_without_interaction_for_volcano <- anova_results_without_interaction %>% 
  mutate(enriched_1 = as.numeric(lfc2 > 0 & qvalue < 0.05),
         enriched_2 = as.numeric(lfc2 < 0 & qvalue < 0.05),
         significant_temp = (enriched_1 + enriched_2 * 2) + 1,
         significant = case_when(group == "tissue" & significant_temp > 1 ~ significant_temp + 2,
                                 group == "tissue" & significant_temp == 1 ~ significant_temp,
                                 group == "genotype" ~ significant_temp),
         significant = as.factor(significant))


anova_results_without_interaction_for_volcano %>%
  ggplot(aes(lfc2, -log10(p.value), color = significant)) +
  geom_point() +
  scale_color_manual("More abundant in:",
                     values = c("black", "#00BFC4", "#F8666D", "#7CAE00", "#C77CFF"),
                     breaks = c(2, 3, 4, 5),
                     labels = c("MIDY", "WT", "Adipose, mesenterial", "Adipose, subcutaneous")) +
  facet_grid(~ group, scales = "free_x") +
  theme(panel.background = element_blank(),
        axis.line = element_line("black"))
```

# Linear models


# t-test

Function for performing multiple t-tests.

```{r echo = FALSE}
multiple_ttests <- function(tb, col_name, equal_var = FALSE) {
  
  # Libraries
  library(tidyverse)
  library(broom)
  
  # Quote
  col_name <- enquo(col_name)
  
  # Calculate the mean for each protein
  means <- tb %>%
    group_by(name) %>%
    summarise(mean_total = mean(LFQ))
  
  
  # ttest
  t_test <-  tb %>%
    dplyr::select(name, !!col_name, LFQ) %>%
    group_by(name, !!col_name) %>%
    nest() %>%
    spread(key = !!col_name, value = data) %>%
    group_by(name) %>%
    mutate(p_value = t.test(unlist(midy), unlist(wt), var.equal = equal_var)$p.value,
           mean_midy = mean(unlist(midy)),
           mean_wt = mean(unlist(wt)),
           log2_difference = mean_midy - mean_wt)
  
  # Join data sets
  t_test <- left_join(t_test, means, by = "name")
  
  return(t_test)
}
```

## Mesenterial

First select mesenterial tissue.

```{r}
mesenterial_data <- adipose_imputed %>%
  dplyr::filter(tissue == "m")
```

### t-test, mesenterial, unequal variance

```{r}
# t_test for mesenterial tissue
t_test_m <- multiple_ttests(mesenterial_data, genotype, equal_var = FALSE)
```

```{r}
plot_pvalue_histogram <- function(tb, col_name, limit_max, plot_title = "p-value histogram") {
  
  # Quote
  col_name <- enquo(col_name)
  
  # pvalue plot
  tb %>% 
    ggplot(aes(x = !!col_name)) +
    geom_histogram(breaks = seq(0, 1, 0.05), fill = "black", color = "white") +
    xlab("p-value") +
    ylab("Number of proteins") +
    scale_y_continuous(limits = c(0, limit_max), expand = c(0, 0)) + 
    labs(
      title = plot_title
    ) +
    theme(panel.background = element_blank(),
          axis.line = element_line(color = "black"))
}
```

Plot p-value histogram.

```{r}
plot_pvalue_histogram(t_test_m, p_value, 150)
```

Use `qvalue` to calculate q-values.

```{r}
fdr_correction_with_qvalue <- function(tb, p_value_col) {
  
  # Quote
  p_value_col <- enquo(p_value_col)
  
  # fdr correction
  qvalues <- qvalue(as_vector(tb %>%
                                ungroup() %>% # To make sure to only get one column
                                dplyr::select(!!p_value_col)))
  
  return(qvalues)
}
```

```{r}
q_values <- fdr_correction_with_qvalue(t_test_m, p_value)

plot(q_values)
hist(q_values)
```

```{r}
join_qvalues_with_results <- function(tb, p_value_col = "p_value") {
  
  # Make a tibble with p and qvalues
  q_values <- tibble(p_value = q_values[["pvalues"]], q_value = q_values[["qvalues"]])
  
  # Join with ANOVA results
  tb <- full_join(tb, q_values, by = p_value_col)
  
  return(tb)
}
```

```{r}
t_test_m_unequal <- join_qvalues_with_results(t_test_m)
```

__MA plot__

```{r}
plot_MA_plot <- function(tb, plot_title = "MA plot") {
  tb %>% 
    ggplot(aes(mean_total, log2_difference)) +
    geom_point() +
    geom_hline(yintercept = 0, color = "red") +
    xlab("Mean intensity") +
    ylab("Log2 fold difference") +
    labs(
      title = plot_title
    ) +
    theme(
      panel.background = element_blank(),
      axis.line = element_line("black")) 
}
```

```{r}
plot_MA_plot(t_test_m_unequal)
```

__Volcano plot__

```{r}
plot_volcano_pvalue <- function(tb, label_1, label_2,
                                plot_title = "Volcano plot") {
  
  # Volcano_plot
  tb %>% 
    mutate(
      enriched_1 = as.numeric(log2_difference > 0.585 & p_value < 0.05),
      enriched_2 = as.numeric(log2_difference < -0.585 & p_value < 0.05),
      significant = as.factor((enriched_1 + enriched_2 * 2) + 1)
    ) %>%
    ggplot(aes(log2_difference, -log10(p_value), color = significant)) +
    geom_hline(yintercept = -log10(0.05)) +
    geom_vline(xintercept = log2(1.5)) +
    geom_vline(xintercept = -log2(1.5)) +
    geom_point() +
    scale_color_manual("More abundant in:",
                       values = c("black", "#00BFC4", "#F8666D"),
                       breaks = c(2, 3),
                       labels = c(label_1, label_2)
    ) +
    labs(
      title = plot_title
    ) +
    xlab(bquote(~ log[2] ~ " fold change")) +
    ylab(bquote("-" ~ log[10] ~ "(p-value)")) +
    #  title = plot_title
    # ) +
    theme(
      panel.background = element_blank(),
      axis.line = element_line("black")
    )
}
```

```{r}
t_test_m_unequal <- t_test_m_unequal %>%
  mutate(enriched_midy = as.numeric(log2_difference > 0.585 & p_value < 0.05),
         enriched_wt   = as.numeric(log2_difference < -0.585 & p_value < 0.05),
         significant   = as.factor((enriched_midy + enriched_wt * 2) + 1))
         

kable(t_test_m_unequal %>%
        ungroup() %>%
        summarise(`Significnt in midy` = sum(log2_difference > 0.585 & p_value < 0.05),
                  `Significant in wt`  = sum(log2_difference < -0.585 & p_value < 0.05)),
      format = "pandoc", align = "c",
      caption = "Significant proteins with log2 fold change < -0.585 or > 0.585 and p-value > 0.05")

  # Volcano plot
 t_test_m_unequal %>%
  mutate(
    enriched_midy = as.numeric(log2_difference > 0.585 & p_value < 0.05),
    enriched_wt = as.numeric(log2_difference < -0.585 & p_value < 0.05),
    significant = as.factor((enriched_midy + enriched_wt * 2) + 1)
  ) %>%
  ggplot(aes(log2_difference, -log10(p_value), color = significant)) +
  geom_point() +
  scale_color_manual("More abundant in:",
    values = c("black", "#00BFC4", "#F8666D"),
    breaks = c(2, 3),
    labels = c("midy", "wt")
  ) +
  xlab(bquote(~ log[2] ~ " fold change")) +
  ylab(bquote("-" ~ log[10] ~ "(p-value)")) +
  #  title = plot_title
  # ) +
  theme(
    panel.background = element_blank(),
    axis.line = element_line("black")
  )
```


```{r}
plot_volcano <- function(tb, plot_title = "Volcano plot") {
  tb %>% 
    ggplot(aes(log2_difference, -log10(p_value))) +
    geom_point() +
    xlab("Log2 fold difference") +
    ylab("-log10(p-value") +
    labs(
      title = plot_title
    ) +
    theme(
      panel.background = element_blank(),
      axis.line = element_line("black")) 
}
```

```{r}
plot_volcano(t_test_m_unequal)
```


### t-test, mesenterial, equal variance

```{r}
# t_test for mesenterial tissue
t_test_m_equal <- multiple_ttests(mesenterial_data, genotype, equal_var = TRUE)
```

```{r}
plot_pvalue_histogram(t_test_m_equal, p_value, 150)
```

Use `qvalue` to calculate q-values.


```{r}
q_values <- fdr_correction_with_qvalue(t_test_m_equal, p_value)

plot(q_values)
hist(q_values)
```

```{r}
t_test_m_equal <- join_qvalues_with_results(t_test_m_equal)
```

### Subcutaneous

First select subcutaneous tissue.

```{r}
sc_data <- adipose_imputed %>%
  dplyr::filter(tissue == "sc")
```

### t-test, sc, unequal variance

```{r}
t_test_sc_unequal <- multiple_ttests(sc_data, genotype, equal_var = FALSE)
```

